routing/localchans/manager.go

```diff
@@ -212,6 +214,13 @@ func (r *Manager) updateEdge(tx kvdb.RTx, chanPoint wire.OutPoint,
 		edge.MinHTLC = *newSchema.MinHTLC
 	}
 
+	// If the channel is unannounced and the dont_forward bit is still not
+	// set we set it here. This is an on-the-fly migration. The edge is
+	// persistet to disk later when propagating the new channel update.
+	if unannounced && !edge.MessageFlags.HasDontForward() {
```

Comment:
Probably missing bit flip mechanics here but why not just set if the channel is unannounced?